---
title: "Writing Prompt 3"
author: "Joshua Bray (18477142)"
date: "April 24, 2018"
output: html_document
---
*"Discuss the challenges involved in defining a microbial species and how HGT complicates matters, especially in the context of the evolution and phylogenetic distribution of microbial metabolic pathways.  Can you comment on how HGT influences the maintenance of global biogeochemical cycles through time?  Finally, do you think it is necessary to have a clear definition of a microbial species?  Why or why not?"*
 
 
####Introduction
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;In his evolutionary book *On the Origin of Species,* Charles Darwin eloquently stated:  
"As many more individuals of each species are born than can possibly survive; and as, consequently, there is a frequently recurring struggle for existence, it follows that any being, if it vary however slightly in any manner profitable to itself, under the complex and sometimes varying conditions of life, will have a better chance of surviving, and thus be naturally selected" [1].  
This statement has defined the paradigm that is evolutionary biology for 150 years. However, the complexity of microbial evolution was not yet known at the time of Darwin's book, for horizontal gene transfer (HGT) would not be discovered for another 60 years [2]. We now have a better understanding of the complexity of microbial evolution as well as the immense diversity of microbial life, but defining exactly what constitutes a microbial species has many complicating factors, such as HGT, that we have yet to definitively clarify and incorporate into a complete microbial species definition.  

&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;There have been many different approaches to classifying microbial species. One of the most common methods of classifying new organisms is a polyphasic approach that takes into consideration multiple characteristics of the microbe like 16S rRNA sequencing, DNA-DNA hybridization, and phenotypic characterizations [3]. While each individual approach has its own weaknesses, collectively they have been used to classify many microbial species. However, with advances in genome sequencing technology it has become apparent that even within a species there can be incredible genomic diversity [4, 5]. This has led to the development of new, genomic classification techniques such as multiple locus sequence analyses (MLSA), which uses the sequences of housekeeping genes to determine phylogenetic relationships [6]. One of the issues with the MLSA approach is determining the threshold or the transition point between species' boundaries that is not subjective, but approaches like genealogical concordance attempt to resolve this issue by objectively delineating groups that are representative of species (Figure 1) [7]. While genealogical concordance may prove to be useful for objectively determining species' boundaries, it only utilizes housekeeping genes and thus avoids the complexity of HGT and has limited ability to distinguish between ecotypes. In other words, it classifies organisms by their vertically evolved gene backbones without taking into account the variable, HGT-derived genomic islands [5].    
 

 ![**Figure 1. Genomic species hypotheses.** A standard MLSA tree creates multiple potential species hypotheses due to the subjective nature of determining the transition point between species' boundaries (a), while genealogical concordance analyses result in objective species' boundaries by determining the transition point between genealogical discord within species and genealogical concordance between species (b). Figure from [7]. ](GenomicSpeciesFigure.png)
 

####Phylogenetic Distribution of Microbial Metabolic Pathways  
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Microbial evolution has been greatly influenced by dynamic community structures and HGT, making it possible for entire metabolic pathways to spread throughout Earth's ecology. For example, proteorhodopsin photosystems can be horizontally-transferred in a single event allowing the recipients to use photophosphorylation to supplement their energy needs [8]. Being able to exchange the incredibly versatile, fitness-improving photosystem with a single HGT event likely explains the ecological and phylogenetic prevalence of these photosystems in nature [8]. However, at the community level it is common to find metabolic networks that are distributed amongst community members rather than equally shared. The reason for this is that there is a fitness advantage for smaller genome sizes, so organisms will reduce their replication costs by losing genes for metabolic pathways that are already supported by another organism [9, 10]. This phenomenon has been referred to as the black queen hypothesis in reference to the card came "Hearts," where the players need to strategically avoid taking the queen of spades by tricking their opponents to take it for themselves [9]. This is analogous to individual microbes passing off the production of essential metabolites to others, which has been seen in marine vitamin synthesis and trafficking [10]. Restricting a community's metabolite production to a limited number of organisms may put the community at risk over geological time scales, but because of the flexibility HGT provides it is possible to maintain those communities when the environmental conditions fluctuate.  

&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;The Earth's biosphere has undergone many microbially-driven changes over time, and not only has HGT been responsible for influencing the Earth's biogeochemical cycles, but HGT has also helped maintain them over the eons. In our oceans today, microbial communities are dominated by a small number of different populations, but there are also thousands of low-abundance populations that are responsible for most of the communities' observed phylogenetic diversity [11]. This "rare biosphere" has persisted over geological time scales and is likely responsible for episodically reshaping Earth's biogeochemical processes [11]. These low-abundance populations can also be viewed as "guardians of metabolism," because there is enough diversity in their populations to survive environmental perturbation and maintain fundamental biogeochemical metabolisms [12]. Also, it is possible for those metabolic systems to undergo extensive HGT and be spread throughout the biosphere and even between domains of life [12]. For example, this has been found to have occurred with the transfer of sulfate respiration between bacteria [13], methane-oxidation from Archaean methanogens to bacteria [14]; and nitrogenases from an Archaean source to cyanobacteria [15]. While all of these examples demonstrate how HGT has influenced and maintained Earth's biogeochemical cycles over time, they also demonstrate how HGT has influenced the evolution and phylogenetic distribution of microbial metabolic pathways.


####Conclusion  
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Defining a microbial species is complicated by HGT. Our current definitions rely on multiple approaches of classification, such as 16S rRNA sequencing, DNA-DNA hybridization, phenotypic classifications, and multiple variations of MLSA analyses [3, 6, 7]. Each method has its drawbacks, which is why polyphasic approaches have been gaining popularity [3], but next generation sequencing (NGS) technologies have made whole genome analyses a practical option [16]. While still in their infancy, NGS technologies have drastically lowered the cost of whole genome sequencing for researchers, but there has yet to be a universally-accepted consensus on how to appropriately analyze the data [16]. A popular method is the average nucleotide identity (ANI) analysis, which uses in silico comparisons of whole genomes to determine the phylogenetic relationship between species [16]. However, this method still has its own share of problems, such as still requiring a subjective threshold for species' boundaries [16], intra-species variations in genomic content due to HGT [17], and approximately 18% prokaryotic species defined by a polyphasic approach suffer from anomalies when analyzed by ANI [18]. Of course, this might not be an issue of ANI being incorrect but rather the polyphasic approach being incorrect, but it serves to demonstrate the complexity of defining microbial species when there is no universally accepted method.  

&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;One thing that must be asked at this point, is whether a clear microbial species definition is even necessary. This is a difficult question to answer, because determining what constitutes as "necessary" is sure to be subjective. I feel that a clear microbial species definition is definitely something that researchers should strive towards, because a universally accepted method would avoid classification anomalies when carrying out meta-analyses as done by Konstantinidis et al. [18]. However, the importance of a clear microbial species definition is threatened by ANI analyses that have found certain genera that do not exhibit clustering, while unrelated, obligatory pathogens that inhabit extremely similar, restricted niches do exhibit clusters [17]. This once again brings up the concept of ecotypes, which are a form of classification focusing on the ecological niches that organisms inhabit, and which can have incredible variation within species [5, 17, 19]. Another important component of niches that must be addressed are microbial communities, because it has been clearly established that microbial communities possess dynamic, distributed metabolic networks [9, 10, 20], so it does not make sense to solely classify a microbe as an individual when it truly exists as a member of a larger community. Perhaps in the future we will have a much more robust polyphasic approach to microbial classification that not only relies upon the individual microbe's genomic and phenotypic characteristics, but also a separate classification of ecotype that can be shared between different microbial species and also encompasses microbial communities.  

####References 
[1] Darwin C. 1859. On the origin of species *by means of natural selection, or the preservation of favoured races in the struggle for life.* London: John Murray. p 5. 
http://graphics8.nytimes.com/packages/images/nytint/docs/charles-darwin-on-the-origin-of-species/original.pdf  
[2] Griffith F. 1928. The Significance of Pneumococcal Types. Journal of Hygiene 27:113-159. 
https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2167760/pdf/jhyg00267-0003.pdf  
[3] Prakash O, Verma M, Sharma P, Kumar M, Kumari K, Singh A, Kumari H, Jit S, Gupta SK, Khanna M, Lal R. 2007. Polyphasic approach of bacterial classification - An overview of recent advances. Indian Journal of Microbiology 47:98-108.  
https://dx.doi.org/10.1007%2Fs12088-007-0022-x  
[4] Thompson JR. 2005. Genotypic Diversity Within a Natural Coastal Bacterioplankton Population. Science 307:1311-1313.
http://science.sciencemag.org/content/307/5713/1311  
[5] Welch RA, Burland V, Plunkett G, Redford P, Roesch P, Rasko D, Buckles EL, Liou S- R, Boutin A, Hackett J, Stroud D, Mayhew GF, Rose DJ, Zhou S, Schwartz DC, Perna NT, Mobley HLT, Donnenberg MS, Blattner FR. 2002. Extensive mosaic structure revealed by the complete genome sequence of uropathogenic Escherichia coli. Proceedings of the National Academy of Sciences 99:17020-17024.
[www.pnas.org/cgi/doi/10.1073/pnas.252529799](www.pnas.org/cgi/doi/10.1073/pnas.252529799)  
[6] Glaeser SP, Kämpfer P. 2015. Multilocus sequence analysis (MLSA) in prokaryotic taxonomy. Systematic and Applied Microbiology 38:237-245.
https://doi.org/10.1016/j.syapm.2015.03.007  
[7] Venter SN, Palmer M, Beukes CW, Chan W-Y, Shin G, Zyl EV, Seale T, Coutinho TA, Steenkamp ET. 2017. Practically delineating bacterial species with genealogical concordance. Antonie van Leeuwenhoek 110:1311-1325.
https://rdcu.be/L71c  
[8] Martinez A, Bradley AS, Waldbauer JR, Summons RE, Delong EF. 2007. Proteorhodopsin photosystem gene expression enables photophosphorylation in a heterologous host. Proceedings of the National Academy of Sciences 104:5590-5595.
[www.pnas.org/cgi/doi/10.1073/pnas.0611470104](www.pnas.org/cgi/doi/10.1073/pnas.0611470104)  
[9] Morris JJ, Lenski RE, Zinser ER. 2012. The Black Queen Hypothesis: Evolution of Dependencies through Adaptive Gene Loss. mBio 3.
http://doi.org/10.1128/mBio.00036-12.  
[10] Giovannoni SJ. 2012. Vitamins in the sea. Proceedings of the National Academy of Sciences 109:13888-13889.
[www.pnas.org/cgi/doi/10.1073/pnas.1211722109](www.pnas.org/cgi/doi/10.1073/pnas.1211722109)   
[11] Sogin ML, Morrison HG, Huber JA, Welch DM, Huse SM, Neal PR, Arrieta JM, Herndl GJ. 2006. Microbial diversity in the deep sea and the underexplored "rare biosphere". Proceedings of the National Academy of Sciences 103:12115-12120.
[www.pnas.org/cgi/doi/10.1073/pnas.0605127103](www.pnas.org/cgi/doi/10.1073/pnas.0605127103)  
[12] Falkowski PG, Fenchel T, Delong EF. 2008. The Microbial Engines That Drive Earths Biogeochemical Cycles. Science 320:1034-1039.
https://DOI.org/10.1126/science.1153213  
[13] Friedrich MW. 2002. Phylogenetic Analysis Reveals Multiple Lateral Transfers of Adenosine-5-Phosphosulfate Reductase Genes among Sulfate-Reducing Microorganisms. Journal of Bacteriology 184:278-289.
https://www.ncbi.nlm.nih.gov/pubmed/11741869  
[14] Chistoserdova L, Vorholt JA, Thauer RK, Lidstom ME. 1998. C1 Transfer Enzymes and Coenzymes Linking Methylotrophic Bacteria and Methanogenic Archaea. Science 281:99-102.
https://www.ncbi.nlm.nih.gov/pubmed/9651254  
[15] Kechris KJ, Lin JC, Bickel PJ, Glazer AN. 2006. Quantitative exploration of the occurrence of lateral gene transfer by using nitrogen fixation genes as a case study. Proceedings of the National Academy of Sciences 103:9584-9589.
https://doi.org/10.1073/pnas.0603534103  
[16] Rosselló-Móra R, Amann R. 2015. Past and future species definitions for Bacteria and Archaea. Systematic and Applied Microbiology 38:209-216. 
https://doi.org/10.1016/j.syapm.2015.02.001  
[17] Konstantinidis KT, Ramette A, Tiedje JM. 2006. The bacterial species definition in the genomic era. Philosophical Transactions of the Royal Society B: Biological Sciences 361:1929-1940.
https://dx.doi.org/10.1098%2Frstb.2006.1920  
[18] Varghese NJ, Mukherjee S, Ivanova N, Konstantinidis KT, Mavrommatis K, Kyrpides NC, Pati A. 2015. Microbial species delineation using whole genome sequences. Nucleic Acids Research 43:6761-6771.
https://doi.org/10.1093/nar/gkv657  
[19] Koeppel A, Perry EB, Sikorski J, Krizanc D, Warner A, Ward DM, Rooney AP, Brambilla E, Connor N, Ratcliff RM, Nevo E, Cohan FM. 2008. Identifying the fundamental units of bacterial diversity: A paradigm shift to incorporate ecology into bacterial systematics. Proceedings of the National Academy of Sciences 105:2504-2509.
https://doi.org/10.1073/pnas.0712205105  
[20] Zaikova E, Walsh DA, Stilwell CP, Mohn WW, Tortell PD, Hallam SJ. 2010. Microbial community dynamics in a seasonally anoxic fjord: Saanich Inlet, British Columbia. Environmental Microbiology 12:172-191.
https://doi.org/10.1111/j.1462-2920.2009.02058.x  